### PR TITLE
`Integrated code lifecycle`: Fix flickers in build queue

### PR DESCRIPTION
**Pull request by BBesrour**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

There is a small issue when build jobs are added/removed from the queue that caused it to flicker which made it hard to keep track of the table.

### Steps for Testing

Prerequisites:
- 1 instructor or admin
- 1 programming exercise

1. Open the build queue (if logged in as instructor, you can find it under course management > build queue in the top right)
2. In another tab, spam the build queue by submitting repeatedly
3. Keep an eye on the build queue and make sure that it's not flickering and working correctly. Also make sure that the sort is working

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **New Features**
  - Improved build queue component to provide more accurate job duration tracking.
  
- **Bug Fixes**
  - Fixed an issue where job durations were not properly calculated in the build queue.
  
- **Tests**
  - Updated unit tests to reflect changes in how build job durations are calculated.

### CODE REVIEW

File: src/main/webapp/app/localci/build-queue/build-queue.component.ts [hunk 1]
```diff
@@ -152,7 +152,7 @@ export class BuildQueueComponent implements OnInit, OnDestroy {
         this.buildStatusFilterValues = Object.values(BuildJobStatusFilter);
         this.loadQueue();
         this.interval = setInterval(() => {
-            this.updateBuildJobDuration();
+            this.runningBuildJobs = this.updateBuildJobDuration(this.runningBuildJobs);
```
[the remark this answers is not in the document]
Comment (BBesrour) [reply src/main/webapp/app/localci/build-queue/build-queue.component.ts:155]: What would the code look like?

**Review by kaancayli — APPROVED** [review]
Tested during testing session on TS3. Works as described. Code LGTM 👍

**Review by DominikRemo — APPROVED** [review]
Tested in testing session on ts3. LGTM

**Review by EneaGore — APPROVED** [review]
Tested in testing session. lgtm

**Review by SimonEntholzer — APPROVED** [review]
Worked fine during testing session :+1:

**Review by coolchock — APPROVED** [review]
tested in testing sessions, LGTM

**Review by JohannesStoehr — COMMENTED** [review]
We also have a pipe specifically for this purpose, called `artemisTimeAgo`. You might think about using this instead of your custom solution

**Review by krusche — APPROVED** [review]

**Pull request merged**